extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesPeonClient.java

```diff
@@ -111,7 +111,8 @@ public JobResponse waitForJobCompletion(K8sTaskId taskId, long howLong, TimeUnit
                           unit
                       );
       if (job == null) {
-        return new JobResponse(job, PeonPhase.FAILED);
+        log.info("K8s job for task was not found %s", taskId);
```

Comment:
let's also log the likely reason for this in the log line itself.

Reply:
@abhishekagarwal87  I wrongly took this to mean you wanted the reason to actually be surfaced in the error message so I added it there in addition to updating this log line.

the error looks like this now
<img width="1209" alt="Screenshot 2023-04-03 at 6 58 23 PM" src="https://user-images.githubusercontent.com/17736581/229644725-cf663bb3-3f67-412d-a62c-e068d9a004df.png">

lmk if you think thats too much info to surface in the console and I can remove that commit, otherwise I addressed the rest of the comments

Reply:
This is better. Though "task was deleted" is more of an implementation detail. "Task was canceled or pod got terminated" seems better. So if pods were reaped then users will know immediately. Or if they canceled the job, they will be able to put two and two together themselves.

Reply:
i put deleted because the task shows as cancelled when you shut it down through the console but failed if you e.g. delete the k8s job